### PR TITLE
feat: adjusting the 'add network screen' in network bottom sheet, for network UI redesign 

### PR DESCRIPTION
**Pull request by EtherWizard33**
## **Description**
This PR shows the 'Add custom network' form directly after clicking the button, where the user can add and connect to a custom network. A demo can be viewed here: https://www.loom.com/share/bd380aca1065428ea584b0421ce69fb9

Here is a link to the figma: https://www.figma.com/design/76R5BvAVubUhWaN2Ld7MAv/Default-Networks?node-id=1303-34400&m=dev

## **Related issues**

Ensures that the form is displayed directly after clicking since the list of popular networks are now integrated in the list.

## **Manual testing steps**

1. From the home page, click the network selector at the top of the screen, a bottom sheet comes up with a list of networks
2. At the bottom of the bottom sheet, click Add custom network button, a form should show up allowing to add a network
3. Fill the form with relevant information and add the network

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/NetworkSelector/NetworkSelector.tsx [hunk 1]
```diff
@@ -200,6 +268,9 @@ const NetworkSelector = () => {
       // TODO: Provide correct types for network.
       const { name, imageSource, chainId } = (Networks as any)[networkType];
 
+      if (isNetworkUiRedesignEnabled && !name.includes(searchString))
```
Comment (darkwing) [inline app/components/Views/NetworkSelector/NetworkSelector.tsx:271]: This looks to be repeated as well.

File: app/components/Views/NetworkSelector/NetworkSelector.tsx [hunk 2]
```diff
@@ -129,8 +144,46 @@ const NetworkSelector = () => {
...
+    return searchResult;
+  };
+
   const renderMainnet = () => {
     const { name: mainnetName, chainId } = Networks.mainnet;
+
+    if (
+      isNetworkUiRedesignEnabled &&
+      filterNetworksByName([Networks.mainnet], searchString).length === 0
+    )
```
Comment (darkwing) [inline app/components/Views/NetworkSelector/NetworkSelector.tsx:184]: Let's make this a helper function where we pass in the network

File: app/components/Views/NetworkSelector/NetworkSelector.tsx (hunk 1 above)
Comment (EtherWizard33) [reply app/components/Views/NetworkSelector/NetworkSelector.tsx:271]: Hi David, good point, this was already added to PR #2 currently in review and the change will be propagated to this current PR soon.

**Comment by salimtb** [comment]
work as expected , good job !

**Review by salimtb — APPROVED** [review]

File: app/components/Views/NetworkSelector/NetworkSelector.tsx [hunk 3]
```diff
@@ -147,7 +150,6 @@ const NetworkSelector = () => {
     }
   };
 
-  // TODO: type the any below to import { Network } from './CustomNetwork.types';
   // TODO: Replace "any" with type
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const showNetworkModal = (networkConfiguration: any) => {
```
Comment (tommasini) [inline app/components/Views/NetworkSelector/NetworkSelector.tsx:155]: We have the goal of removing the any instances of the codebase to make it better! Do you think you could remove and type the anys of this file and remove the disable at the top?

**Review by tommasini — APPROVED** [review]
LGTM! Approving since the any was no introduced on this PR but if you have the opportunity to remove it, it would be awesome !

File: app/components/Views/NetworkSelector/NetworkSelector.tsx (hunk 3 above)
Comment (EtherWizard33) [reply app/components/Views/NetworkSelector/NetworkSelector.tsx:155]: yes, I noticed the PR that enabled the rule a few days ago.
I have updated some of the 'any' already, I plan to do this one too, 
next PR is also in this file, I will look into it :)

**Pull request merged**